float64 min_num
float64 max_num
